### ISSUE TITLE
WorkspaceChange: createFile,  renameFile, deleteFile fail without other edits

### ISSUE THREAD

**Issue by wiseman** [microsoft/vscode-languageserver-node#696]
This code fails with the error `Workspace edit is not configured for document changes`:

```typescript
const workspaceChange = new WorkspaceChange();
const uri = "file:///Users/wisej041/test.json";
workspaceChange.createFile(uri, {overwrite: true});
```

I think it's because `createFile` is defined like this:
```typescript
public createFile(uri: DocumentUri, options?: CreateFileOptions): void {
	this.checkDocumentChanges();
	this._workspaceEdit!.documentChanges!.push(CreateFile.create(uri, options));
}

private checkDocumentChanges() {
	if (!this._workspaceEdit || !this._workspaceEdit.documentChanges) {
		throw new Error('Workspace edit is not configured for document changes.');
	}
}
```

It seems that if the `WorkspaceChange` doesn't already have `documentChanges`, then I can't use `createFile` to add another change? I don't quite understand how this is supposed to be used.

**Comment by dbaeumer** [comment]
Good catch

**Comment by dbaeumer** [comment]
I fixed that for the 3.16 release.
